apps/partners-reference/lib/formatApplicationData.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  ApplicationUpdate,
+  ApplicantUpdate,
+  Language,
+  IncomePeriod,
+  ApplicationSubmissionType,
+  ApplicationStatus,
+  Address,
+} from "@bloom-housing/core"
+import moment from "moment"
+
+/*
+  Some of fields are optional, not active, so it occurs 'undefined' as value.
+  This function eliminates those fields and parse to a proper format.
+*/
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+
+function getAddress(condition, addressData) {
+  const blankAddress: Address = {
+    id: undefined,
+    createdAt: undefined,
+    updatedAt: undefined,
+    street: "",
+    street2: "",
+    city: "",
+    state: "",
+    zipCode: "",
+  }
+
+  return condition ? addressData : blankAddress
+}
+
+export const formatApplicationData = (data: any, listingId, editMode: boolean) => {
+  const language: Language = data.application.language ? data.application.language : null
+
+  const submissionDate: Date | null = (() => {
+    // rename default (wrong property names)
+    const { birthDay: submissionDay, birthMonth: submissionMonth, birthYear: submissionYear } =
+      data.dateSubmitted || {}
+    const { hours, minutes = 0, seconds = 0, time } = data.timeSubmitted || {}
+
+    if (!submissionDay || !submissionMonth || !submissionYear) return null
+
+    const date = new Date()
+
+    date.setUTCDate(submissionDay)
+    date.setUTCMonth(submissionMonth - 1)
+    date.setUTCFullYear(submissionYear)
+
+    if (hours && minutes && seconds && time) {
```

Comment:
minutes = 0 and seconds = 0, despite being correct values will cause skipping this if branch and will set time to `      date.setUTCHours(0, 0, 0, 0)`, right?